jupyterhub/remoteappmanager_config.py

```diff
@@ -29,7 +29,8 @@
 # Define the sqlalchemy url for the database.
 # Notes:
 # 1. this database is shared among all instances of the remoteappmanager.
-# 2. if a relative path is specified, it will be relative to the working
-#    directory specified at startup, if specified, otherwise to the cwd
-#    at startup. Note that jupyterhub spawners spawn in the home directory.
-db_url = "sqlite:///"+os.path.expanduser("~/remoteappmanager.db")
+# 2. When running with system-user mode, the jupyterhub spawners spawn in
+#    the user's home directory; when running in virtual-user mode,
+#    the current directory is the directory where jupyterhub is started
+# 3. '~' would be evaluated as the spawned user's home directory
+db_url = "sqlite:///"+os.path.abspath("./remoteappmanager.db")
```

Comment:
> if we add one --option or an envvar, I don't want to run the risk that we update in one and not the other...

That's true.  Instead of having a base class, have a base test case?